wirez-extensions/wirez-lienzo-extensions/src/main/java/org/wirez/lienzo/grid/Grid.java

```diff
@@ -55,17 +59,39 @@ public Point findPosition(Point anchorPoint, Direction direction) {
                 x -= width;
                 y -= height;
                 break;
+            default:
+                throw new UnsupportedOperationException();
         }
 
         return new Point(x, y);
     }
 
     public Point findPosition(int row, int col) {
-        int x = padding + (col * (padding + iconSize));
-        int y = padding + (row * (padding + iconSize));
+        if (!isInRange(row, getRows())) {
+            throw new IllegalArgumentException(
+                    String.format("%s is incorrect row value. Value have to be from 0 to %s", row, getRows() - 1)
```

Comment:
I think it's neither emulated...http://www.gwtproject.org/doc/latest/RefJreEmulation.html